systemd/emulation.go

```diff
@@ -150,13 +150,22 @@ func (s *emulation) EnsureMountUnitFile(description, what, where, fstype string)
 		return mountUnitName, nil
 	}
 
-	hostFsType, actualOptions := hostFsTypeAndMountOptions(fstype)
+	// Create directory as systemd would do when enabling the unit
+	if err := os.MkdirAll(filepath.Join(dirs.GlobalRootDir, unitOptions.Where), 0755); err != nil {
+		return "", err
+	}
+
+	// In emulation mode hostFsType is the fs we want to use to manually mount
+	// the snap below, but fstype is used for the created mount unit.
+	// This means that when preseeding in a lxd container, the snap will be
+	// mounted with fuse, but mount unit will use squashfs.
+	hostFsType, actualOptions := hostFsTypeAndMountOptions(unitOptions.Fstype)
```

Comment:
this not using unitOptions.Options which seems a bit strange?

Reply:
True, but here we need to distinguish between the options for the unit file that will be stored in the preseeding tarball, and that should match what is passed to the method, and the options for mounting temporarily a snap so it is available to create the tarball. This is for the second case, and trying to apply the "real" options might be problematic as we are in a different host than the target. Note also that this is similar to what previously `EnsureMountUnitFile` was doing for emulation mode, where the options for the unit are different to those for the mount call.

Reply:
I think we might want a comment about this though,  because as you said it might be immaterial but might also bite us, so some sort of comment or TODO would be appropriate for the next person wondering the same

Reply:
Agreed, I've added additional comments to clarify this.